rmf_fleet_adapter/src/full_control/main.cpp

```diff
@@ -467,11 +468,31 @@ struct Connections : public std::enable_shared_from_this<Connections>
           path_request_pub, mode_request_pub);
 
     const auto& l = state.location;
+    const auto& starts = rmf_traffic::agv::compute_plan_starts(
+        *graph, state.location.level_name, {l.x, l.y, l.yaw},
+        rmf_traffic_ros2::convert(adapter->node()->now()));
+    if (starts.empty())
+    {
+      RCLCPP_ERROR(
+        adapter->node()->get_logger(),
+        "Unable to compute a StartSet for robot [%s] using level_name [%s] and "
+        "location [%f, %f, %f] specified in its RobotState message. This can "
+        "happen if the level_name in the RobotState message does not match any "
+        "of the map names in the navigation graph supplied or if the location "
+        "reported in the RobotState message is far way from the navigation "
+        "graph. This robot will not be added to the fleet [%s].",
+        state.name.c_str(),
+        state.location.level_name.c_str(),
+        l.x, l.y, l.yaw,
+        fleet_name.c_str());
+
+        throw std::runtime_error(
```

Comment:
This isn't what I meant when I suggested throwing an exception. I think it's fine to just let it continue from here and ignore the robot in the future.

Effectively what I meant is that [this assertion](https://github.com/open-rmf/rmf_ros2/blob/888a35a418a623e9e063857b7d54d3cd18fdbd50/rmf_fleet_adapter/src/rmf_fleet_adapter/agv/FleetUpdateHandle.cpp#L987) should be promoted to throwing an exception, so the robot adding routine always halts there no matter what mode it was built in.

Reply:
Aaah I apologise for misunderstanding your comment. I hope the latest change reflects what you had in mind.